storage/src/nodestore.rs

```diff
@@ -524,47 +590,54 @@ impl ReadInMemoryNode for Committed {
     }
 }
 
-#[derive(Debug)]
+#[derive(Clone, Debug)]
 pub enum NodeStoreParent {
     Proposed(Arc<ImmutableProposal>),
-    Committed,
+    Committed(Option<TrieHash>),
 }
 
-impl<S: ReadableStorage> From<NodeStore<Committed, S>> for NodeStoreParent {
-    fn from(_: NodeStore<Committed, S>) -> Self {
-        NodeStoreParent::Committed
-    }
-}
-
-impl<S: ReadableStorage> From<NodeStore<ImmutableProposal, S>> for NodeStoreParent {
-    fn from(val: NodeStore<ImmutableProposal, S>) -> Self {
-        NodeStoreParent::Proposed(Arc::new(val.kind))
+impl PartialEq for NodeStoreParent {
```

Comment:
I believe this also meets the requirements for `Eq`

Reply:
fixed